BuildScript.Nuke/BaseBuild.Compile.cs

```diff
@@ -56,59 +70,90 @@ public partial class BaseBuild : NukeBuild
       .DependsOn(RestoreReleaseBuild, RestoreTestBuild);
 
   protected Target RestoreReleaseBuild => _ => _
-      .DependsOn(ReadConfiguration)
+      .DependsOn(ReadConfiguration, CleanFolders)
       .Unlisted()
-      .Executes(() =>
+      .Executes(async () =>
       {
-        ReleaseProjectFiles.ForEach(RestoreProject);
+        await RestoreProjects(ReleaseProjectFiles);
       });
 
 
   protected Target RestoreTestBuild => _ => _
-      .DependsOn(ReadConfiguration)
+      .DependsOn(ReadConfiguration, CleanFolders)
       .Unlisted()
-      .Executes(() =>
+      .Executes(async () =>
       {
-        TestProjectFiles.ForEach(RestoreProject);
+        await RestoreProjects(TestProjectFiles);
       });
 
-  private void CompileProject (IReadOnlyCollection<ProjectMetadata> projectFiles)
+  private async Task CompileProjects (IReadOnlyCollection<ProjectMetadata> projectFiles)
   {
-    MSBuild(s => s
-        .SetAssemblyVersion(SemanticVersion.AssemblyVersion)
-        .SetFileVersion(SemanticVersion.AssemblyFileVersion)
-        .SetCopyright(AssemblyMetadata.Copyright)
-        .SetProperty(MSBuildProperties.PackageVersion, SemanticVersion.AssemblyNuGetVersion)
-        .SetProperty(MSBuildProperties.CompanyName, AssemblyMetadata.CompanyName)
-        .SetProperty(MSBuildProperties.CompanyUrl, AssemblyMetadata.CompanyUrl)
-        .SetProperty(MSBuildProperties.ProductName, AssemblyMetadata.ProductName)
-        .SetProperty(MSBuildProperties.AssemblyOriginatorKeyFile, Directories.SolutionKeyFile)
-        .DisableRestore()
-        .When(GitRepository != null, s => s.SetPackageProjectUrl(GitRepository!.HttpsUrl))
-        .CombineWith(projectFiles, (settings, projectFile) =>
-            settings.SetToolsVersion(projectFile.ToolsVersion)
-                .SetTargetPath(projectFile.ProjectPath)
-                .SetTargets(GetCompileTargets(projectFile))
-                .SetConfiguration(projectFile.Configuration)
-                .SetInformationalVersion(SemanticVersion.GetAssemblyInformationalVersion(projectFile.Configuration, AdditionalBuildMetadata))
-        )
-    );
+    var projectFilesDebugTargets = projectFiles
+        .Where(file => file.Configuration == "Debug").ToList();
+    var projectFilesReleaseTargets = projectFiles
+        .Where(file => file.Configuration == "Release").ToList();
+    if (projectFilesDebugTargets.Count > 0) await CompileProject(projectFilesReleaseTargets, "Debug", "Current");
+    
+    if (projectFilesReleaseTargets.Count > 0) await CompileProject(projectFilesReleaseTargets, "Release", "Current");
```

Comment:
The tools version from the other PR should be used here I guess. (Instead of `Current`). Just a reminder to adapt the second PR in case this gets merged first.

Reply:
The other one got merged first, so please adapt :)